### PR TITLE
Add GeneralRuleViewModelTest

### PR DESCRIPTION
**Pull request by juliewww**
---
name: Pull request
about: Create a pull request
label: 'triage me'
---
Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way
  we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter
  pass (`./gradlew --init-script gradle/init.gradle.kts spotlessApply` to automatically apply
  formatting)
- [ ] Appropriate docs were updated (if necessary)

Is this your first Pull Request?

- [ ] Run `./tools/setup.sh`
- [ ] Import the code formatting style as explained in [the setup script](/tools/setup.sh#L40).

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: core/domain/build.gradle.kts [hunk 1]
```diff
@@ -32,6 +32,7 @@ dependencies {
     implementation(projects.core.rule)
 
     testImplementation(projects.core.testing)
+    implementation(libs.androidx.lifecycle.livedata.ktx)
```
Comment (lihenggui) [inline core/domain/build.gradle.kts:35]: I don't know the reason for importing a new dependency here

File: core/domain/src/main/kotlin/com/merxury/blocker/core/domain/InitializeRuleStorageUseCase.kt [hunk 2]
```diff
@@ -48,22 +48,25 @@ class InitializeRuleStorageUseCase @Inject constructor(
...
-            CopyRulesToStorageWorker.copyWork(),
-        )
-        workManager.getWorkInfosByTagFlow(CopyRulesToStorageWorker.WORK_NAME)
-            .collect { workInfoList ->
-                workInfoList.forEach { workInfo ->
-                    Timber.d("WorkInfo: ${workInfo.tags}, state = ${workInfo.state}")
-                    when (workInfo.state) {
-                        State.SUCCEEDED -> emit(InitializeState.Done)
-                        else -> emit(InitializeState.Initializing(workInfo.state.name))
+        workManager.apply {
```
Comment (lihenggui) [inline core/domain/src/main/kotlin/com/merxury/blocker/core/domain/InitializeRuleStorageUseCase.kt:52]: No need to make changes here

File: feature/generalrule/build.gradle.kts [hunk 3]
```diff
@@ -24,11 +24,14 @@ android {
     namespace = "com.merxury.blocker.feature.generalrule"
 }
 dependencies {
+    implementation(libs.androidx.work.ktx)
```
Comment (lihenggui) [inline feature/generalrule/build.gradle.kts:27]: new dependency?

Comment (juliewww) [reply feature/generalrule/build.gradle.kts:27]: For mock<WorkManager>()

File: core/domain/build.gradle.kts (hunk 1 above)
Comment (juliewww) [reply core/domain/build.gradle.kts:35]: For using asFlow()

Comment (lihenggui) [reply core/domain/build.gradle.kts:35]: Is itnecessary to do it?

File: feature/generalrule/build.gradle.kts (hunk 3 above)
Comment (lihenggui) [reply feature/generalrule/build.gradle.kts:27]: You can remove this now

**Review by lihenggui — APPROVED** [review]

**Pull request merged**